File: example/src/test/java/specs/ui/Ui.java
package specs.ui;

import org.openqa.selenium.By;
import specs.Specs;

import static com.codeborne.selenide.Condition.exist;
import static com.codeborne.selenide.Condition.text;
import static com.codeborne.selenide.Selenide.$;

public class Ui extends Specs {

    public void hasText(String text) {
        $(By.tagName("span")).shouldHave(text(text));
    }

    public String noParamsCheck() {
        $(By.tagName("span")).should(exist);
        return "valueFromMethodCall";
    }

    public boolean areEqual(String s1, String s2) {
        return s1.equals(s2);
    }
}

File: example/src/test/kotlin/specs/ui/Ui.kt
package specs.ui

import com.codeborne.selenide.Condition.exist
import com.codeborne.selenide.Condition.text
import com.codeborne.selenide.Selenide.`$`
import org.openqa.selenium.By
import specs.Specs

class UiPlugin : Specs()
class Ui : Specs() {
    fun hasParagraphText(text: String) {
        `$`(By.tagName("p")).shouldHave(text(text))
    }

    fun noParamsCheck(): String {
        `$`(By.tagName("p")).should(exist)
        return "valueFromMethodCall"
    }

    fun areEqual(s1: String, s2: String): Boolean {
        return s1 == s2
    }
}
